### PR TITLE
Twilson63/feat hyper connect create 312

### PR DESCRIPTION
**Pull request by twilson63**
WIP - started hyper-connect dual nodejs/deno module

### CODE REVIEW

File: packages/connect/package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "@hyper.io/connect",
+  "version": "1.0.0",
```
Comment (TillaTheHun0) [inline packages/connect/package.json:3]: I would make the version sub `1.0.0` until it implemented all of the services.

File: packages/connect/package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "@hyper.io/connect",
```
Comment (TillaTheHun0) [inline packages/connect/package.json:2]: was the original client published under `@hyper`? Any reason not to use a `@hyper` scope? These changes can also wait until we actually intent to publish.

**Review by TillaTheHun0 — COMMENTED** [review]
LGTM. Though just two questions about `package.json`

File: packages/connect/package.json (hunk 2 above)
Comment (twilson63) [reply packages/connect/package.json:2]: We do not own the `@hyper` npm org, we have `@hyper63` or `@hyper.io`

File: packages/connect/package.json (hunk 1 above)
Comment (twilson63) [reply packages/connect/package.json:3]: yep, changed to `0.0.1`

Comment (TillaTheHun0) [reply packages/connect/package.json:3]: Did you push this up? I'm not seeing the version change.

File: packages/connect/deps.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+let crocks, R, signJWT, jwt;
+
+if (globalThis.Deno) {
+  crocks = (await import("https://cdn.skypack.dev/crocks")).default;
+  R = await import("https://cdn.skypack.dev/ramda");
+  signJWT = (await import("https://deno.land/x/djwt@v2.1/mod.ts")).create;
+} else {
+  crocks = (await import("crocks")).default;
+  R = (await import("ramda")).default;
+  jwt = (await import("jsonwebtoken"));
```
Comment (TillaTheHun0) [inline packages/connect/deps.js:10]: I believe top level import is only supported in node 16. Do we want to restrict the use of hyper connect to node 16 and up?

File: packages/connect/connect.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  const br = buildRequest(cs);
+
+  const $ = (svc, client, action, ...args) => {
+    return client[action](...args).runWith(br(svc)).toPromise();
+  };
+
+  /**
+   * @param {string} domain
+   */
+  return function (_domain = "default") {
```
Comment (TillaTheHun0) [inline packages/connect/connect.js:17]: Domain will eventually need to be incorporated for use in hyper cloud.

**Review by TillaTheHun0 — COMMENTED** [review]
Code looks good, but had a couple questions.

I see a lot in PR from previous merges into `main`. It might be worth rebasing to remove those files from this PR.

File: packages/connect/deps.js (hunk 3 above)
Comment (twilson63) [reply packages/connect/deps.js:10]: It should be noted in the package.json to be restricted to v16

File: packages/connect/connect.js (hunk 4 above)
Comment (twilson63) [reply packages/connect/connect.js:17]: Yes, once cloud.hyper.io is ready, then we can incorporate domain.

**Comment by twilson63** [comment]
I would prefer to just merge, I don't like removing history, but if keeping true history bothers you that much I will rebase.

But it does remove the truth imo

**Review by TillaTheHun0 — APPROVED** [review]
LGTM. Awesome work.

**Pull request merged**